File: core/script/locale.py
# $Id: locale.py 784 2008-02-19 15:01:42Z SFB $

from java.text import NumberFormat
from java.util import Locale

import sys, os

try: True
except:
    True = 1
    False = not True


def programName(): return os.path.splitext(os.path.split(sys.argv[0])[1])[0]


def usage():

    return "Usage: %s [<language> [<country> [<variant>]]]" % programName()


def main(argv):

    if len(argv) > 4: return usage()

    if len(argv) == 2 and argv[1] == '?':
        print "Available locales:"
        for locale in Locale.getAvailableLocales():
            print "\t %s" % locale
        return

    if len(argv) > 1:
        language = argv[1]
        if len(argv) > 2: country = argv[2]
        else: country = ''
        if len(argv) > 3: variant = argv[3]
        else: variant = ''
        locale = Locale(language, country, variant)
    else: locale = Locale.getDefault()

    print "Locale: %s" % locale
    numberFormat = NumberFormat.getInstance(locale)
    decimalSymbols = numberFormat.getDecimalFormatSymbols()
    print "Grouping: '%s'" % decimalSymbols.getGroupingSeparator()


if __name__ == '__main__': sys.exit(main(sys.argv))


# End.

File: core/script/locale.js
// $Id: locale.js 2713 2015-07-28 18:18:05Z SFB $

'use strict';

var System = Java.type('java.lang.System');
var NumberFormat = Java.type('java.text.NumberFormat');
var Locale = Java.type('java.util.Locale');

function _fail(message, code) {
    print(message);
    quit(code);
}

function _usage() {
    _fail("Usage: locale [<language> [<country> [<variant>]]]", 1);
}

if (arguments.length > 3) _usage();

if (arguments.length == 1 && arguments[0] == '?') {
    System.out.println("Available locales:");
    for each (locale in Locale.getAvailableLocales()) {
        System.out.printf("\t %s\n", locale);
    }
    quit();
}

var locale;

if (arguments.length > 0) {
    language = arguments[0];
    if (arguments.length > 1) country = arguments[1];
    else country = '';
    if (arguments.length > 2) variant = arguments[2];
    else variant = '';
    locale = new Locale(language, country, variant);
} else locale = Locale.getDefault();

System.out.printf("Locale: %s\n", locale);

var numberFormat = NumberFormat.getInstance(locale);
var decimalSymbols = numberFormat.getDecimalFormatSymbols();

System.out.printf("Grouping: '%s'\n",
    String.fromCharCode(decimalSymbols.getGroupingSeparator()));

// End.
